source/develop/release-management/features/virt/high-performance-vm.html.md

```diff
@@ -0,0 +1,176 @@
+This feature should be implemented the same as the CPU pinning is implemented. 
+Ideally the the I/O Thread pinning should be done automatically and not manually by the user, But in first phase we will not support it.  The automatic pinning can be done by pinning to the CPU that is responsible for the IO in the host (and don't present this CPU to the guest) or alternatively pinned to CPU#0 as this is typically the one. 
+
+In case the I/O Thread pinning topology is not set upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the I/O thread to a host and verify it fits the host configuration.
+Consider verifying if the  I/O is bound to specific CPU on host level and chose that CPU for pinning the I/O Thread.
+
+In addition, we will suggest to set the emulatorpin to be pinned to the same CPU(s) as the I/O Thread. Should consider how the user will do that (using a hook or via VDSM).
+
+#### **Enable virtual NUMA and set NUMA pinning topology**
+Displayed in 'Host' side-tab of the VM dialog.
+
+In case the virtual NUMA and the NUMA pinning topology are not set upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to pin the VM to a host that supports NUMA topology and verify the VM NUMA topology fits the host's NUMA topology.
+
+Ideally the NUMA pinning should be done automatically according to the exposed host's NUMA topology and not manually by the user, but in first phase we will not support it.
+
+#### **disable kernel same page merging (KSM)**
+This is currently implemented in engine only for cluster level (new/edit cluster dialog->'Optimization side-tab). 
+
+KSM can be manually deactivated in host level by stopping the ksmtuned and the ksm service on the hypervisor.
+
+Disabling the KSM for the specific VM may be implemented for oVirt 4.3 if required. This should be implemented by adding a check-box to the 'Resource Allocation' side-tab of the VM dialog and implemented by setting the "nosharepages" domain property in libvirt xml (requires  VDSM changes).
+
+In case that KSM in cluster level or host level is not disabled upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to disable it either in cluster level, host level or for the specific VM.
+
+#### **Enable  memory backing with huge pages**
+This is currently not implemented. 
+This feature includes host side changes in addition to engine and UI changes. The UI changes will be added as a new key option in the 'Custom Properties' side-tab of the VM dialog. 
+This new "hugepages" key will allow to set the size of the huge page in kB (as configured at Hypervisor Boot Time).
+
+In case that huge pages configuration is not set upon VM saving, a recommendation/warning will be displayed in a pop-up and the user will be asked to set it and verify it fits the host configuration. 
+
+For more information please see https://trello.com/c/ABUiJgWR/62-hugepages
+
```

Comment:
We do enable virtio-rng by default (since 'Blank' template default is declared as such), but since you can also edit an existing VM to become high performance, you need to remember enabling it anyway.
The list here includes the required configuration for HP regardless to VM creation defaults and current state.

For enabling virtio-rng you need to set "Period duration" value in UI or leave empty and use libvirt default. Since we enable it automatically, we'll use libvirt defaults (unless you think we should change it).
From https://libvirt.org/formatdomain.html: 
Random number generator device - rate:
The optional rate element allows limiting the rate at which entropy can be consumed from the source. The mandatory attribute bytes specifies how many bytes are permitted to be consumed per period. An optional period attribute specifies the duration of a period in milliseconds; if omitted, the period is taken as 1000 milliseconds (1 second). Since 1.0.4